File: Examples/Java/Containers/MongoDB-REST/src/main/java/com/api/mongodb/rest/controller/TelemetryController.java
package com.api.mongodb.rest.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import com.api.mongodb.rest.service.mongodb.crud.MongoDB_CRUD_Client;
import com.api.mongodb.rest.telemetry.Telemetry;
import org.bson.BsonDocument;
import org.bson.BsonInt64;
import org.bson.types.ObjectId;

import java.time.Instant;
import java.time.Duration;
import java.time.LocalDateTime;
import java.time.ZoneOffset;

@RestController
@RequestMapping("/api/telemetry")
public class TelemetryController {
    @PostMapping("/{raw_start}")
    public ResponseEntity<Telemetry> postTelemetry(@PathVariable double raw_start, @RequestBody String requestBody) throws Exception {
        Instant start = Instant.ofEpochSecond((long) raw_start)
                .plusMillis((long) ((raw_start % 1) * 1000));
        double containerTime = Instant.now().toEpochMilli() / 1000.0;
        Duration sslHandshake = Duration.between(start, Instant.now());
        Instant start_ref = Instant.now();

        Instant measure = Instant.now();
        BsonDocument body = BsonDocument.parse(requestBody);
        Duration deserialization = Duration.between(measure, Instant.now());

        String db_name = "test";
        String collection_name = "source";
        measure = Instant.now();
        MongoDB_CRUD_Client client = new MongoDB_CRUD_Client();
        ObjectId document_id = client.createAsync(db_name, collection_name, body).get();
        Duration createObject = Duration.between(measure, Instant.now());

        measure = Instant.now();
        BsonDocument document = client.readAsync(db_name, collection_name, document_id).get();
        Duration getObject = Duration.between(measure, Instant.now());

        measure = Instant.now();
        String jsonData = document.toJson();
        Duration serialization = Duration.between(measure, Instant.now());

        measure = Instant.now();
        for (int i = 0; i < 15; i++) {
            client.updateAsync(db_name, collection_name, document_id, new BsonDocument("time", new BsonInt64(Instant.now().toEpochMilli()))).get();
        }
        Duration highIOps = Duration.between(measure, Instant.now());

        measure = Instant.now();
        client.deleteAsync(db_name, collection_name, document_id).get();
        Duration deleteObject = Duration.between(measure, Instant.now());

        Duration total = Duration.between(start_ref, Instant.now());

        Telemetry telemetry = new Telemetry();
        telemetry.setStart(raw_start);
        telemetry.setContainerTime(containerTime);
        telemetry.setSslHandshake(sslHandshake.toNanos() / 1_000_000.0);
        telemetry.setDeserialization(deserialization.toNanos() / 1_000_000.0);
        telemetry.setCreateObject(createObject.toNanos() / 1_000_000.0);
        telemetry.setGetObject(getObject.toNanos() / 1_000_000.0);
        telemetry.setSerialization(serialization.toNanos() / 1_000_000.0);
        telemetry.setHighIOps(highIOps.toNanos() / 1_000_000.0);
        telemetry.setDeleteObject(deleteObject.toNanos() / 1_000_000.0);
        telemetry.setTotal(total.toNanos() / 1_000_000.0);

        return ResponseEntity.ok(telemetry);
    }
}

File: Examples/PHP/Containers/Laravel-MongoDB-REST/src/app/Http/Controllers/TelemetryController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use App\Services\MongoDB\CRUD;
use MongoDB\Model\BSONDocument;

class TelemetryController extends Controller
{
    public function index(Request $request, $raw_start) { 
        $start = (float)$raw_start;
        $containerTime = microtime(true);
        $sslHandshake = $containerTime - $start;
        $start_ref = microtime(true);

        $measure = microtime(true);
        $data = $request->getContent();
        $body = new BSONDocument(json_decode($data, true));
        $deserialization = microtime(true) - $measure;

        $db_name = "test";
        $collection_name = "source";
        $measure = microtime(true);
        $client = new CRUD\MongoDB_CRUD_Client();
        $document_id = $client->createAsync($db_name, $collection_name, $body);
        $createObject = microtime(true) - $measure;

        $measure = microtime(true);
        $document = $client->readAsync($db_name, $collection_name, $document_id);
        $getObject = microtime(true) - $measure;

        $measure = microtime(true);
        $jsonData = json_encode($document);
        $serialization = microtime(true) - $measure;

        $measure = microtime(true);
        for ($i = 0; $i < 15; $i++) {
            $json_update = '{"time": "' . date('Y-m-d H:i:s') . '"}';
            $client->updateAsync($db_name, $collection_name, $document_id, new BSONDocument(json_decode($json_update, true)));
        }
        $highIOps = microtime(true) - $measure;

        $measure = microtime(true);
        $client->deleteAsync($db_name, $collection_name, $document_id);
        $deleteObject = microtime(true) - $measure;

        $total = microtime(true) - $start_ref;

        $telemetry = [
            'start' => $raw_start,
            'container_time' => $containerTime,
            'ssl_handshake' => $sslHandshake,
            'deserialization' => $deserialization * 1000,
            'create_object' => $createObject * 1000,
            'get_object' => $getObject * 1000,
            'serialization' => $serialization * 1000,
            'high_iops' => $highIOps * 1000,
            'delete_object' => $deleteObject * 1000,
            'total' => $total * 1000
        ];

        return response()->json($telemetry, 200);
    }
}